### PR TITLE
gql schema skeleton

### PR DESCRIPTION
**Pull request by jaismith**
# gql schema skeleton

first stab at a data schema for our api, will need to support full feature parity with current LL, still need to figure out exactly how relations 
will work through appsync but everything in here i _believe_ is supported by default.

see schema/src/*.graphql

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "name": "layuplist",
   "version": "1.0.0",
   "repository": "https://github.com/layuplist/ll2",
-  "author": "Jai Smith, Adam McQuilkin",
-  "license": "MIT"
+  "author": "dplanner-dev",
+  "license": "MIT",
+  "workspaces": ["infra", "schema", "client"],
+  "private": true
```
Comment (ajmcquilkin) [inline package.json:8]: This isn't compatible with the license

File: schema/dist/types.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
+export type Maybe<T> = T | undefined;
```
Comment (ajmcquilkin) [inline schema/dist/types.ts:1]: Are these the current DB schemas converted to TS or are there updates?

File: schema/dist/types.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+export type InputMaybe<T> = T | undefined;
```
Comment (ajmcquilkin) [inline schema/dist/types.ts:2]: Do we have null checking enabled in the TSConfig? Also, how is this different from the `Maybe<T>`?

File: schema/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,103 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
+    /* Projects */
+    // "incremental": true,                              /* Save .tsbuildinfo files to allow for incremental compilation of projects. */
+    // "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
+    // "tsBuildInfoFile": "./.tsbuildinfo",              /* Specify the path to .tsbuildinfo incremental compilation file. */
+    // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects. */
+    // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
```
Comment (ajmcquilkin) [inline schema/tsconfig.json:10]: Should we wipe the commented fields?

**Review by ajmcquilkin — DISMISSED** [review]
Looks super solid! Just a few minor comments

File: package.json (hunk 1 above)
Comment (jaismith) [reply package.json:8]: needs to be private to be used as a yarn workspace root.

File: schema/dist/types.ts (hunk 2 above)
Comment (jaismith) [reply schema/dist/types.ts:1]: these are generated from the gql types, which i completely made up without even looking at the current ll schemas. not final whatsoever.

File: schema/dist/types.ts (hunk 3 above)
Comment (jaismith) [reply schema/dist/types.ts:2]: graphql distinguishes inputs from normal types, i assume this is a util type codegen is creating to represent those (even though i've not defined any yet). we can tweak how the gql -> typescript transpilation happens, but we will
need input types in some form to use gql clients.

File: schema/tsconfig.json (hunk 4 above)
Comment (jaismith) [reply schema/tsconfig.json:10]: yes.

Comment (jaismith) [reply schema/tsconfig.json:10]: updated

**Pull request merged**